src/masternode-sync.cpp

```diff
@@ -324,12 +324,12 @@ void CMasternodeSync::ProcessTick()
         if(Params().NetworkIDString() == CBaseChainParams::REGTEST)
         {
             if(nRequestedMasternodeAttempt <= 2) {
-                pnode->PushMessage(NetMsgType::GETSPORKS); //get current network sporks
+                g_connman->PushMessageWithVersion(pnode, INIT_PROTO_VERSION, NetMsgType::GETSPORKS); //get current network sporks
```

Comment:
`PushMessageWithVersion` does not send `somemessage` and `version` messages, it sends `somemessage` serializing it for `someversion`. I think this shouldn't cause any issues.

Re checking `nVersion`: yes, we can check either `nVersion` or `nSendVersion` to make sure we received `version` message. But as it was said above it's not smth that this PR breaks or trying to fix, it's already how things work in current design. It was simply discovered here that things do not work as they should, so fixing this should probably be done in a separate PR that fixes exactly this issue.